### PR TITLE
Rework rct_vehicle_colour

### PR DESCRIPTION
**Pull request by duncanspumpkin**
This will break replays.

### CODE REVIEW

File: src/openrct2/core/DataSerialiserTraits.h [hunk 1]
```diff
@@ -749,22 +749,24 @@ template<> struct DataSerializerTraits_t<TrackDesignSceneryElement>
...
+        stream->Read(&val.Body);
+        stream->Read(&val.Trim);
+        stream->Read(&val.Ternary);
     }
-    static void log(OpenRCT2::IStream* stream, const rct_vehicle_colour& val)
+    static void log(OpenRCT2::IStream* stream, const VehicleColour& val)
     {
         char msg[128] = {};
-        snprintf(msg, sizeof(msg), "rct_vehicle_colour(body_colour = %d, trim_colour = %d)", val.body_colour, val.trim_colour);
+        snprintf(msg, sizeof(msg), "VehicleColour(body = %d, trim = %d, tertiary = %d)", val.Body, val.Trim, val.Ternary);
```
Comment (tupaschoal) [inline src/openrct2/core/DataSerialiserTraits.h:769]: Ternary or tertiary?

Comment (Broxzier) [reply src/openrct2/core/DataSerialiserTraits.h:769]: > Ternary: Made up of three things; treble, triadic, triple, triplex
> Tertiary: Of third rank or order; subsequent. 

Tertiary seems better.

Comment (duncanspumpkin) [reply src/openrct2/core/DataSerialiserTraits.h]: Okay i've swapped it to tertiary. Annoyingly the plugin api uses ternary but its not the first nore the last where the plugin api uses an old name.

**Review by tupaschoal — COMMENTED** [review]
Seems fine, though I am a bit confused as to why it breaks replays (less fields being saved/read?)

**Comment by duncanspumpkin** [comment]
Yeah reorg of elements will break replays. To be expected. Will get that sorted later

**Comment by duncanspumpkin** [comment]
I've remade this PR could someone review then I will record new replays.

**Pull request merged**